scipy/stats/stats.py

```diff
@@ -5297,12 +5329,14 @@ def kstest(rvs, cdf, args=(), N=20, alternative='two-sided', mode='approx'):
           * 'two-sided'
           * 'less': one-sided, see explanation in Notes
           * 'greater': one-sided, see explanation in Notes
-    mode : {'approx', 'asymp'}, optional
+    mode : {'auto', 'exact', 'approx', 'asymp'}, optional
         Defines the distribution used for calculating the p-value.
-        The following options are available (default is 'approx'):
+        The following options are available (default is 'auto'):
```

Comment:
is the mode only relevant for 'two-sided'?

Reply:
Yes.  The one-sided tests call  `distributions.ksone.sf` and that decides when to switch from exact to an approximation.

Reply:
should we mention this in the documentation?